src/repository/http.rs

```diff
@@ -228,10 +228,10 @@ where
             if status == StatusCode::NOT_FOUND {
                 Err(Error::NotFound)
             } else {
-                Err(Error::Opaque(format!(
-                    "Error getting {:?}: {:?}",
-                    self.uri, resp
-                )))
+                Err(Error::BadHttpStatus {
+                    code: status,
+                    uri: uri.to_string(),
+                })
             }
         } else {
             Ok(resp)
```

Comment:
WRT the intent of this change, it's worth noting the `status.is_success` check accepts _any_ 200-level response, which is not correct per the expected semantics of the request. We should probably be checking specifically for `StatusCode::OK` and forwarding anything else here.